### PR TITLE
next app

### PR DESCRIPTION
**Pull request by ankur-dwivedi**

### CODE REVIEW

File: next/app/page.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+"use client";
+import React, { useRef } from "react";
+import { IKImage, ImageKitProvider, IKUpload, IKVideo } from "imagekit-next";
+
+const publicKey = "<YOUR_IMAGEKIT_PUBLIC_KEY>";
+const urlEndpoint = "<YOUR_IMAGEKIT_URL_ENDPOINT>";
```
Comment (eclectic-memory) [inline next/app/page.js:6]: Move all config to a single file so that it's easier for user to provide the config.

**Review by eclectic-memory — CHANGES_REQUESTED** [review]

File: next/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+npm run dev
+# or
+yarn dev
+# or
+pnpm dev
+# or
+bun dev
+```
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/README.md:15]: removed it

File: next/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Getting Started
+
+First, run the development server:
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/README.md:5]: done

File: next/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+This project uses [`next/font`](https://nextjs.org/docs/basic-features/font-optimization) to automatically optimize and load Inter, a custom Google Font.
+
+## Learn More
+
+To learn more about Next.js, take a look at the following resources:
+
+- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
+- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+
+You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/README.md:30]: removed it

File: next/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Deploy on Vercel
+
+The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+
+Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/README.md:36]: removed

File: next/app/api/auth/route.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  const expire = searchParams.get('expire') || (Math.floor(Date.now() / 1000) + 2400).toString();
+  const privateAPIKey = `${privateKey}`;
+  const signature = crypto.createHmac('sha1', privateAPIKey).update(token + expire).digest('hex');
+
+  return NextResponse.json({
+    token: token,
+    expire: expire,
+    signature: signature
+  });
+}
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/app/api/auth/route.ts:21]: done updated it

File: next/app/page.js (hunk 1 above)
Comment (ankur-dwivedi) [reply next/app/page.js:6]: Done, moved to `.env.local`

File: next/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  "name": "next",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (ankur-dwivedi) [reply next/package.json:11]: As discussed on call, it won't be required.

File: next/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+```bash
+npm run dev
+```
+
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+You can start editing the page by modifying `app/page.js`. The page auto-updates as you edit the file.
+
+This project uses [`next/font`](https://nextjs.org/docs/basic-features/font-optimization) to automatically optimize and load Inter, a custom Google Font.
```
Comment (eclectic-memory) [inline next/README.md:25]: Remove this line. Not needed.

File: next/.env.example [hunk 9]
```diff
@@ -0,0 +1,3 @@
+NEXT_PUBLIC_PUBLIC_KEY = "<YOUR_IMAGEKIT_PUBLIC_KEY>"
+NEXT_PUBLIC_URL_ENDPOINT = "<YOUR_IMAGEKIT_URL_ENDPOINT>"
+PRIVATE_KEY = <YOUR_IMAGEKIT_PRIVATE_KEY>
```
Comment (eclectic-memory) [inline next/.env.example:3]:
```suggestion
NEXT_PUBLIC_PUBLIC_KEY="<YOUR_IMAGEKIT_PUBLIC_KEY>"
NEXT_PUBLIC_URL_ENDPOINT="<YOUR_IMAGEKIT_URL_ENDPOINT>"
PRIVATE_KEY="<YOUR_IMAGEKIT_PRIVATE_KEY>"
```

File: next/package.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    "next": "14.2.4",
+    "react": "^18",
+    "react-dom": "^18"
+  },
+  "devDependencies": {
```
Comment (eclectic-memory) [inline next/package.json:16]: Can we add typescript in dev deps here too so that we can avoid any version conflict or compatibility issues?

**Review by eclectic-memory — APPROVED** [review]

File: next/README.md (hunk 8 above)
Comment (ankur-dwivedi) [reply next/README.md]: Done, removed it.

File: next/.env.example (hunk 9 above)
Comment (ankur-dwivedi) [reply next/.env.example]: Updated it.

File: next/package.json (hunk 10 above)
Comment (ankur-dwivedi) [reply next/package.json:16]: We are not using typescript in this project

Comment (eclectic-memory) [reply next/package.json:16]: @ankur-dwivedi `next-env.d.ts` is still there.

**Comment by eclectic-memory** [comment]
PR is approved from my end. 

Shouldn't be merged until
- The Next.JS integration package is published.
- Zip file is removed from here and `package.json` is updated.

File: next/README.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
Comment (ahnv) [inline next/README.md:1]: Add the line here also 
"for the scope for this example, we will focus on app router in nextjs. Please refer to [here](https://github.com/imagekit/imagekit-docs/-/blob/dev/page_router_example_in_sdk_repo) for page router"

**Review by ahnv — APPROVED** [review]

File: next/README.md (hunk 11 above)
Comment (ankur-dwivedi) [reply next/README.md]: Done.

**Pull request merged**